orchestrator/gorc/src/commands/tx/cosmos.rs

```diff
@@ -87,6 +115,105 @@ impl Runnable for SendToEth {
         let cosmos_address = cosmos_key.to_address("//TODO add to config file").unwrap();
 
         println!("Sending from Cosmos address {}", cosmos_address);
+        let config = APP.config();
+        let cosmos_prefix = config.cosmos.prefix.clone();
+        let cosmso_grpc = config.cosmos.grpc.clone();
+
+        abscissa_tokio::run(&APP, async {
+            let connections =
+                create_rpc_connections(cosmos_prefix, Some(cosmso_grpc), None, TIMEOUT).await;
+            let contact = connections.contact.unwrap();
+            let mut grpc = connections.grpc.unwrap();
+
+            let res = grpc
+                .denom_to_erc20(DenomToErc20Request{
+                    denom: denom.clone(),
+                })
+                .await;
+                match res {
+                    Ok(val) => println!(
+                        "Asset {} has ERC20 representation {}",
+                        denom,
+                        val.into_inner().erc20
+                    ),
+                    Err(_e) => {
+                        println!(
+                            "Asset {} has no ERC20 representation, you may need to deploy an ERC20 for it!",
+                            denom.clone()
+                        );
+                        exit(1);
+                    } }
+                    let amount = Coin {
+                        amount,
+                        denom: denom.clone(),
+                    };
+                    let bridge_fee = Coin {
+                        denom: denom.clone(),
+                        amount: 1u64.into(),
+                    };
+                    let eth_dest: EthAddress = to_eth_addr.parse().unwrap();
+                    check_for_fee_denom(&denom, cosmos_address, &contact).await;
+
+                    let balances = contact
+            .get_balances(cosmos_address)
+            .await
+            .expect("Failed to get balances!");
+        let mut found = None;
+        for coin in balances.iter() {
+            if coin.denom == denom {
+                found = Some(coin);
+            }
+        }
+        println!("Cosmos balances {:?}", balances);
+
+        let times = if let Some (x)=self.times{
+            x
+        } else {
+            1
+        };
+
+        if found.is_none() {
+            panic!("You don't have any {} tokens!", denom);
+        } else if amount.amount.clone() * times.into() >= found.clone().unwrap().amount
+            && times == 1
```

Comment:
First of all, what is the "times" parameter for?

Second, this conditional is odd and overcomplicated. `times == 1 && amount.amount.clone() >= found.clone().unwrap().amount` seems like it would be cleaner.

Reply:
It seems to be there for testing. @jackzampolin - should we keep it?